app/services/RestaurantService.java

```diff
@@ -1,257 +1,331 @@
-		RestaurantReview restaurantReview = (RestaurantReview) getSession().createCriteria(RestaurantReview.class)
-				.add(Restrictions.eq("restaurantId", reviewForm.getRestaurantId()))
-				.add(Restrictions.eq("userId", user.getId()))
-				.uniqueResult();
-		if (restaurantReview == null) {
-			restaurantReview = new RestaurantReview(
-					reviewForm.getRestaurantId(),
-					user.getId(),
-					reviewForm.getReviewScore(),
-					reviewForm.getReviewText()
-			);
-		} else {
-			restaurantReview.setReview(reviewForm.getReviewText());
-			restaurantReview.setRating(reviewForm.getReviewScore());
-		}
+	List<RestaurantReview> restaurantReviews = getSession().createCriteria(RestaurantReview.class)
+		.add(Restrictions.eq("userId", userId)).list();
 
-		getSession().save(restaurantReview);
-		return true;
-	}
+	List<UUID> userReviewRestaurantIds = restaurantReviews.stream().map(RestaurantReview::getRestaurantId)
+		.collect(Collectors.toList());
+	if (userReviewRestaurantIds.size() > 0)
+	{
 
-	/**
-	 * Gets number of restaurants.
-	 *
-	 * @return the number of restaurants
-	 */
-	public Long getNumberOfRestaurants() {
-		return Long.valueOf(getSession().createCriteria(Restaurant.class)
-				.setProjection(Projections.rowCount())
-				.uniqueResult().toString());
-	}
+	    Double temp = (Double) getSession().createCriteria(Restaurant.class)
+		    .add(Restrictions.in("id", userReviewRestaurantIds)).setProjection(Projections.avg("priceRange"))
+		    .uniqueResult();
 
-	/**
-	 * Update picture string.
-	 *
-	 * @param imageUploadForm the image upload form
-	 * @return the string
-	 * @throws Exception the exception
-	 */
-	public String updatePicture(final ImageUploadForm imageUploadForm) throws Exception {
-		Restaurant restaurant = (Restaurant) getSession().createCriteria(Restaurant.class)
-				.add(Restrictions.eq("id", imageUploadForm.getRestaurantId()))
-				.uniqueResult();
-
-		String newImagePath = AWS_BASE_PATH + imageUploadForm.getRestaurantId() + "-" + imageUploadForm.getImageType() + "." + imageUploadForm.getExtension();
-
-		if (imageUploadForm.getImageType().equals("profile")) {
-			restaurant.setProfileImagePath(newImagePath);
-		} else {
-			restaurant.setCoverImagePath(newImagePath);
-		}
+	    Integer priceRange = temp.intValue();
 
-		getSession().update(restaurant);
-		return "{ \"imageFor\": \"" + imageUploadForm.getImageType() + "\", \"url\": \"" + newImagePath + "\"}";
+	    UserHabitsBean userHabitsBean = (UserHabitsBean) getSession()
+		    .createCriteria(RestaurantReview.class, "review").add(Restrictions.eq("userId", userId))
+		    .setProjection(
+			    Projections.projectionList().add(Projections.groupProperty("review.userId").as("userId"))
+				    .add(Projections.avg("review.rating").as("userRatingRange")))
+		    .setResultTransformer(Transformers.aliasToBean(UserHabitsBean.class)).uniqueResult();
+
+	    userHabitsBean.setUserPriceRange(priceRange);
+	    userHabitsBean.setCity(user.getCity().getId());
```

Comment:
This still relies on the users registered city, instead of their actual location.

Reply:
Odakle da uzmem njegovu trenutnu lokaciju, gdje se cuva taj podatak?

Reply:
možeš tražiti od browser-a da, ako user to dozvoli, dobiješ njegovu trenutnu lokcaiju